topcoffea/modules/corrections.py

```diff
@@ -21,13 +21,26 @@ def get_method1a_wgt_singlewp(eff,sf,passes_tag):
     wgt = p_data/p_mc
     return wgt
 
+def get_method1a_wgt_doublewp(effA, effB, sfA, sfB, cutA, cutB, cutC):
+    effA_data = effA * sfA
+    effB_data = effB * sfB
+
+    pMC = ak.prod(effA[cutA], axis=-1) * ak.prod(effB[cutB] - effA[cutB], axis=-1) * ak.prod(1 - effB[cutC], axis=-1)
+    pMC = ak.where(pMC==0,1,pMC) # removeing zeroes from denominator...
+    pData = ak.prod(effA_data[cutA], axis=-1) * ak.prod(effB_data[cutB] - effA_data[cutB], axis=-1) * ak.prod(1 - effB_data[cutC], axis=-1)
+
+    return pData, pMC
+
 # Evaluate btag sf from central correctionlib json
 def btag_sf_eval(jet_collection,wp,year,method,syst):
-
     # Get the right sf json for the given year
-    if year == "2016APV":
+    clib_year = year
+
+    if year.startswith("2016"):
+        clib_year = "2016preVFP" if year == "2016APV" else "2016postVFP"
+    if year == "2016preVFP":
         fname = topcoffea_path("data/btag_sf_correctionlib/2016preVFP_UL_btagging.json")
-    elif year == "2016":
+    elif year == "2016postVFP":
```

Comment:
@anpicci I made these changes because it was complaining that `clib_year` was used before it was defined. Can you just double check I didn't break anything? The CI runs here and on TopEFT, so it's probably correct.

Reply:
@anpicci can you confirm I didn't change the logic here?